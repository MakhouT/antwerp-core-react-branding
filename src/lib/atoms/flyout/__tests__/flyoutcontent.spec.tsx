import * as React from "react";

import * as A from "../../../";
import { renderAndMatchSnapshot } from "../../../../__tests__/__fixtures__/testUtils";

describe("FlyoutContent", () => {
    it("Renders default", () => {
        // Arrange
        const element = (<A.FlyoutContent visible />);

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders with child", () => {
        // Arrange
        const element = (<A.FlyoutContent visible></A.FlyoutContent>);

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders with children", () => {
        // Arrange
        const element = (<A.FlyoutContent visible>
            <p>Child 1</p>
            <p>Child 2</p>
        </A.FlyoutContent>);

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Visible and non visible should be different", () => {
        // Arrange
        const visible = (<A.FlyoutContent visible><p>child</p></A.FlyoutContent>);
        const hidden = (<A.FlyoutContent><p>child</p></A.FlyoutContent>);

        // Act and assert
        const visibletree = renderAndMatchSnapshot(visible);
        const hiddentree = renderAndMatchSnapshot(hidden);

        expect(visibletree).not.toEqual(hiddentree);
    });
});
