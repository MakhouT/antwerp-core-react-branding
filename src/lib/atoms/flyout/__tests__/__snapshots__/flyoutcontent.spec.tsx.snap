// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlyoutContent Renders default 1`] = `
<div
  className="m-flyout__content"
  style={Object {}}
/>
`;

exports[`FlyoutContent Renders with child 1`] = `
<div
  className="m-flyout__content"
  style={Object {}}
/>
`;

exports[`FlyoutContent Renders with children 1`] = `
<div
  className="m-flyout__content"
  style={Object {}}
>
  <p>
    Child 1
  </p>
  <p>
    Child 2
  </p>
</div>
`;

exports[`FlyoutContent Visible and non visible should be different 1`] = `
<div
  className="m-flyout__content"
  style={Object {}}
>
  <p>
    child
  </p>
</div>
`;

exports[`FlyoutContent Visible and non visible should be different 2`] = `<span />`;
